From e1029d5ebf0fb1ea49cd4a26eb6f444f703d6e63 Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Tue, 20 Nov 2012 02:31:18 +0200
Subject: [PATCH 17/22] Fix: disabled name cannot be copied to clipboard, make
 it unfocusable instead.

Set name insensitive is good in visual but we cannot 'Select All'+'Copy'
on it. Therefore make it unfocusable and ineditable instead.
---
 src/gtk/fm-file-properties.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/gtk/fm-file-properties.c b/src/gtk/fm-file-properties.c
index a87b546..39b402b 100644
--- a/src/gtk/fm-file-properties.c
+++ b/src/gtk/fm-file-properties.c
@@ -794,7 +794,8 @@ static void update_ui(FmFilePropData* data)
         strftime(buf, sizeof(buf), "%x %R", &tm);
         gtk_label_set_text(data->atime, buf);
         /* FIXME: changing file name isn't implemented yet, disable entry */
-        gtk_widget_set_sensitive(GTK_WIDGET(data->name), FALSE);
+        gtk_widget_set_can_focus(GTK_WIDGET(data->name), FALSE);
+        gtk_editable_set_editable(GTK_EDITABLE(data->name), FALSE);
     }
     else
     {
-- 
1.8.0.1

